### PR TITLE
feat(server): update date fields of requests, GraphQL request resolvers

### PR DESCRIPTION
**Pull request by mimoham24**
# Overview
- update `UpdateAt` on request update
- update `ApprovedAt` and `ClosedAt` on state change
- fix reviewers resolver
- add item resolver to RequestItem

### CODE REVIEW

File: server/internal/adapter/gql/resolver_request.go [hunk 1]
```diff
@@ -25,13 +25,35 @@ func (r requestResolver) Project(ctx context.Context, obj *gqlmodel.Request) (*g
 }
 
 func (r requestResolver) Reviewers(ctx context.Context, obj *gqlmodel.Request) ([]*gqlmodel.User, error) {
-	res, err := dataloaders(ctx).User.LoadAll(obj.ReviewersID)
-	if len(err) > 0 {
-		return nil, err[0]
-	}
-	return res, nil
+	res, errors := dataloaders(ctx).User.LoadAll(obj.ReviewersID)
+	return res, errors[0]
```
Comment (rot1024) [inline server/internal/adapter/gql/resolver_request.go:29]: Unnecessary change: if errors is nil or empty, it panics

File: server/schemas/request.graphql [hunk 2]
```diff
@@ -25,6 +25,8 @@ type RequestItem {
   version: String
   ref: String
   item: VersionedItem
+  schema: Schema
+  model: Model
```
Comment (rot1024) [inline server/schemas/request.graphql:29]:
```suggestion
```

Add these fields and resolvers to the Item type instead of adding it here.

**Review by rot1024 — CHANGES_REQUESTED** [review]

File: server/schemas/request.graphql (hunk 2 above)
Comment (mimoham24) [reply server/schemas/request.graphql:29]: Do you mean the versioned Item?

File: server/internal/adapter/gql/resolver_request.go (hunk 1 above)
Comment (mimoham24) [reply server/internal/adapter/gql/resolver_request.go:29]: `errors` always [nil]

File: server/schemas/request.graphql (hunk 2 above)
Comment (rot1024) [reply server/schemas/request.graphql:29]: No, `Item` type

File: server/internal/adapter/gql/resolver_request.go (hunk 1 above)
Comment (rot1024) [reply server/internal/adapter/gql/resolver_request.go]: Future implementation changes may cause an error to be returned. There is no particular harm in handling errors, so they should be left as they are.

**Review by rot1024 — APPROVED** [review]

**Review by rot1024 — APPROVED** [review]
